api/v1alpha1/nginxingresscontroller_types.go

```diff
@@ -56,26 +56,38 @@ type NginxIngressControllerSpec struct {
 	// +optional
 	DefaultSSLCertificate *DefaultSSLCertificate `json:"defaultSSLCertificate,omitempty"`
 
+	// DefaultBackendService defines the service that the NginxIngressController should default to when given HTTP traffic with not matching known server names.
+	// The controller directs traffic to the first port of the service. Service should expose /health and / endpoints
+	// +optional
+	DefaultBackendService *NICNamespacedName `json:"defaultBackendService,omitempty"`
+
 	// Scaling defines configuration options for how the Ingress Controller scales
 	// +optional
 	Scaling *Scaling `json:"scaling,omitempty"`
 }
 
 // DefaultSSLCertificate holds a secret in the form of a secret struct with name and namespace properties or a key vault uri
-// +kubebuilder:validation:MaxProperties=1
+// +kubebuilder:validation:MaxProperties=2
 // +kubebuilder:validation:XValidation:rule="(isURL(self.keyVaultURI) || !has(self.keyVaultURI))"
+// +kubebuilder:validation:XValidation:rule="((self.forceSSLRedirect == true) && (has(self.secret) || has(self.keyVaultURI)) || (self.forceSSLRedirect == false))"
 type DefaultSSLCertificate struct {
 	// Secret is a struct that holds the name and namespace fields used for the default ssl secret
 	// +optional
-	Secret *Secret `json:"secret,omitempty"`
+	Secret *NICNamespacedName `json:"secret,omitempty"`
 
 	// Secret in the form of a Key Vault URI
 	// +optional
 	KeyVaultURI *string `json:"keyVaultURI"`
+
+	// ForceSSLRedirect is a flag that sets the global value of redirects to HTTPS if there is a defined DefaultSSLCertificate
+	// +kubebuilder:default:=false
+	// forceSSLRedirect is set to false by default and will add the "forceSSLRedirect: false" property even if the user doesn't specify it.
+	// If a user adds both a keyvault uri and secret the property count will be 3 since forceSSLRedirect still automatically gets added thus failing the check.
+	ForceSSLRedirect bool `json:"forceSSLRedirect,omitempty"`
 }
 
-// Secret is a struct that holds a name and namespace to be used in DefaultSSLCertificate
-type Secret struct {
+// NICNamespacedName is a struct that holds a name and namespace with length checking on the crd
+type NICNamespacedName struct {
```

Comment:
we can't rename this, it would be a breaking change to the CRD (not backwards compatible).

Just make a new struct